scripts/pre_commit_linter.py

```diff
@@ -2421,6 +2430,77 @@ def _check_extra_js_files(self):
 
         return summary_messages
 
+    def _check_constants_declaration(self):
+        """Checks the declaration of constants in the TS files to ensure that
+        the constants are not declared in files other than *.constants.ts.
+        """
+
+        if self.verbose_mode_enabled:
+            print 'Starting constants declaration check'
+            print '----------------------------------------'
+
+        summary_messages = []
+        failed = False
+
+        with _redirect_stdout(_TARGET_STDOUT):
+            all_ts_files = [
+                filepath for filepath in self.all_filepaths if (
+                    filepath.endswith('.ts'))]
+            ts_files_to_check = [
+                filepath for filepath in all_ts_files if not any(
+                    pattern in filepath for pattern in (
+                        OLD_CONVENTION_DIRECTORIES)) and filepath.endswith(
+                            'App.ts')]
+            constants_number_dict = {}
+            for filepath in ts_files_to_check:
+                # Check that the constants are declared only in a *.constants.ts
+                # file.
+                if not filepath.endswith('.constants.ts'):
+                    for line_num, line in enumerate(FileCache.readlines(
+                            filepath)):
+                        if 'oppia.constant(' in line:
+                            failed = True
+                            print ('%s --> Constant declaration found at line '
+                                   '%s. Please declare the constants in a '
+                                   'separate constants file.' % (
+                                       filepath, line_num))
+
+                # Check if the constant has multiple declarations which is
```

Comment:
This is the part where it checks for multiple declarations.

Reply:
Make sure to update the function docstring please.

Reply:
Done.